internal/metadataproviders/aws/ec2/metadata.go

```diff
@@ -30,54 +30,42 @@ type Provider interface {
 }
 
 type metadataClient struct {
-	metadata            *ec2metadata.EC2Metadata
-	metadataRetryEnable *ec2metadata.EC2Metadata
+	metadata               *ec2metadata.EC2Metadata
+	metadataFallbackEnable *ec2metadata.EC2Metadata
 }
 
 var _ Provider = (*metadataClient)(nil)
 
 func NewProvider(sess *session.Session) Provider {
 	return &metadataClient{
 		metadata: ec2metadata.New(sess, &aws.Config{
-			Retryer:                   override.IMDSRetryer,
+			Retryer:                   override.NewIMDSRetryer(override.DefaultIMDSRetries),
 			EC2MetadataEnableFallback: aws.Bool(false),
```

Comment:
This called by datadog and resource detector. It only has telemetry setting as config. I only wanted to configure aws settings. so this can remain a default of 1.